docs/variant_merging.md

```diff
@@ -1,3 +1,227 @@
+  ],
+  [
+    name: "S1",
+    genotype: [0, 1]
+    phaseset: "*"
+    GQ: 48
+  ],
+  [
+    name: "S2",
+    genotype: [0, 0]
+    phaseset: null
+    GQ: 30
+  ]
+}
+DP: 14  # This can also be 11.
+AA: "G"
+```
+
+* With `--copy_quality_to_calls` and `--copy_filter_to_calls`:
+
+```
+reference_name: "1"
+start_position: 100
+end_position: 101
+reference_bases: "A"
+alternate_bases: {alt: "T"}
+names: ["rs1"]
+quality: 29
+filter: ["PASS", "q10"]
+call: {
+  [
+    name: "S3",
+    genotype: [1, 1]
+    GQ: 2
+    quality: 7
+    filter: ["q10"]
+  ],
+  [
+    name: "S1",
+    genotype: [0, 1]
+    phaseset: "*"
+    GQ: 48
+    quality: 29
+    filter: ["PASS"]
+  ],
+  [
+    name: "S2",
+    genotype: [0, 0]
+    phaseset: null
+    GQ: 30
+    quality: 29
+    filter: ["PASS"]
+  ]
+}
+DP: 14  # This can also be 11.
+AA: "G"
+```
+
+* With `--info_keys_to_move_to_calls_regex ^DP$`, `--copy_quality_to_calls`,
+  `--copy_filter_to_calls`:
+
+```
+reference_name: "1"
+start_position: 100
+end_position: 101
+reference_bases: "A"
+alternate_bases: {alt: "T"}
+names: ["rs1"]
+quality: 29
+filter: ["PASS", "q10"]
+call: {
+  [
+    name: "S3",
+    genotype: [1, 1]
+    GQ: 2
+    quality: 7
+    filter: ["q10"]
+    DP: 11
+  ],
+  [
+    name: "S1",
+    genotype: [0, 1]
+    phaseset: "*"
+    GQ: 48
+    quality: 29
+    filter: ["PASS"]
+    DP: 14
+  ],
+  [
+    name: "S2",
+    genotype: [0, 0]
+    phaseset: null
+    GQ: 30
+    quality: 29
+    filter: ["PASS"]
+    DP: 14
+  ]
+}
+AA: "G"
+```
+
+## MERGE_WITH_NON_VARIANTS strategy [EXPERIMENTAL]
+
+This strategy is not yet production ready and can't be used by users. This
+section is included here to provide more context about our ongoing work
+([Issue #80](https://github.com/googlegenomics/gcp-variant-transforms/issues/80)
+and [Issue #88](https://github.com/googlegenomics/gcp-variant-transforms/issues/88)).
+
+The `MOVE_TO_CALLS` strategy is effective for merging calls that have exactly
+the same reference and alternate bases. However, this strategy is not effective
+for merging
+[gVCF files](https://gatkforums.broadinstitute.org/gatk/discussion/4017/what-is-a-gvcf-and-how-is-it-different-from-a-regular-vcf)
+since non-variants are represented as intervals in the genome.
+
+Consider the following VCF file:
+
+
+*`gs://mybucket/c.g.vcf`*:
+
+```
+## <headers omitted>
+#CHROM  POS   ID    REF   ALT        QUAL  FILTER  INFO     FORMAT  S4
+1       50    .     T     <NON_REF>  .     .       END=150  GT:GQ   0/0:40
+```
+
+If you specify `--input_pattern gs://mybucket/*.vcf` with `MOVE_TO_CALLS` merge
+strategy, the non-variant region from call `S4` will not be merged with calls
+`S1`, `S2`, and `S3`. However, we know that call `S4` has genotype `0/0` in
+chr1:100 since it matches the reference from position 50-150 in chr1.
+The `MERGE_WITH_NON_VARIANTS` strategy will infer `0/0` for call `S4` at
+chr1:100 so that the resulting BigQuery table has a single row with all calls
+`S1-4`.
```

Comment:
In this case, do we want to break the [50, 150] segment above into [50, 99] and [101, 150]? Or do we keep the original [50, 150] region (hence S4 will have two entries for 1:100 which are consistent)?

Reply:
Logic is TBD :) Ideally, we should break it into 3 sections (what our existing code tries to do actually), but it gets very complicated, so I'm thinking for the initial version we'd just output SNPs and totally ignore non-variants.
Anyway, I don't want to go into too much detail here as users can't use this feature now anyway.